.gitlab-ci.yml

```diff
@@ -10,127 +16,95 @@
-  BUILD_JOBS: 32
-  NCPUS: 12
-  VERBOSE_BUILD: "OFF"
-  VERBOSE_TEST: "OFF"
-  ON_LASSEN: "ON"
-  ON_QUARTZ: "ON"
-  ON_CORONA: "ON"
-  SPACK_PREFIX: "v0.19.1"
-  SHARED_SPACK: "UPSTREAM"
-  BENCHMARK: "OFF"
-  BENCHMARK_NNODES: 4
-  BENCHMARK_QUEUE: "pbatch"
-
-# Normally, stages are blocking in Gitlab. However, using the keyword "needs" we
-# can express dependencies between job that break the ordering of stages, in
-# favor of a DAG.
-# In practice q_*, l_* and c_* stages are independently run and start immediately.
 
-stages:
-  - q_build_and_test
-  - l_build_and_test
-  - l_build_and_bench
-  - c_build_and_test
+##### PROJECT VARIABLES
+# We build the projects in the CI clone directory.
+# Used in script/gitlab/build_and_test.sh script.
+# TODO: add a clean-up mechanism.
+  BUILD_ROOT: ${CI_PROJECT_DIR}
+  SPACK_REF: "594a376c521cc746978571b1181a47bbcff30a21" # v0.22.2
 
-# These are also templates (.name) that define project specific build commands.
-# If an allocation exist with the name defined in this pipeline, the job will
-# use it (slurm specific).
-.build_toss_3_x86_64_ib_script:
-  script:
-    - echo ${ALLOC_NAME}
-    - srun -p pdebug -N 1 -n ${NCPUS} --interactive -t ${DEFAULT_TIME}
-      --job-name=${ALLOC_NAME} .gitlab/build_and_test.sh
+##### SHARED_CI CONFIGURATION
+# Required information about GitHub repository
+  GITHUB_PROJECT_NAME: "sundials"
+  GITHUB_PROJECT_ORG: "LLNL"
+# Set the build-and-test command.
+# Nested variables are allowed and useful to customize the job command. We
+# prevent variable expansion so that you can define them at job level.
+  JOB_CMD:
+    value: ".gitlab/build_and_test.sh"
+    expand: false
+# Override the pattern describing branches that will skip the "draft PR filter
+# test".  Add protected branches here. See default value in
+# preliminary-ignore-draft-pr.yml.
+#  ALWAYS_RUN_PATTERN: ""
 
-# Corona
-.build_toss_4_x86_64_ib_corona_script:
-  script:
-    - echo ${ALLOC_NAME}
-    - flux alloc -N 1 -t ${DEFAULT_TIME}  .gitlab/build_and_test.sh
+# We organize the build-and-test stage with sub-pipelines. Each sub-pipeline
+# corresponds to a test batch on a given machine.
 
-# CORAL systems use spectrum LSF instead of SLURM
-.build_blueos_3_ppc64le_ib_script:
-  script:
-    - echo ${ALLOC_NAME}
-    - bsub -q pdebug -J ${ALLOC_NAME} -nnodes 1 -W ${DEFAULT_TIME} -Is .gitlab/build_and_test.sh
+# High level stages
+stages:
+  - prerequisites
+  - build-and-test
 
-# Benchmark job for CORAL systems
-.build_blueos_3_ppc64le_ib_bench:
-  script:
-    - echo ${ALLOC_NAME}
-    - bsub -q ${BENCHMARK_QUEUE} -J ${ALLOC_NAME} -nnodes ${BENCHMARK_NNODES} -W ${DEFAULT_TIME} -Is .gitlab/build_and_bench.sh
+# Template for jobs triggering a build-and-test sub-pipeline:
+.build-and-test:
+  stage: build-and-test
+  trigger:
+    include:
+      - local: '.gitlab/custom-jobs-and-variables.yml'
+      - project: 'sundials/radiuss-shared-ci' # https://lc.llnl.gov/gitlab/sundials/radiuss-shared-ci
```

Comment:
In the docs it would be helpful to clarify that `.build-and-test` pulls in files from a mirror of our fork of `radiuss-shared-ci`.

Reply:
Also it looks like you had to add dane to the radiuss-shared-ci repo, will this be pushed back upstream?

Reply:
The PR to upstream it is already open [here](https://github.com/LLNL/radiuss-shared-ci/pull/38).